.github/workflows/main.yml

```diff
@@ -0,0 +1,135 @@
+name: CI
+on:
+  push:
+    branches-ignore: [master]
+  pull_request:
+    branches: ['*']
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  # Check Code style quickly by running `rustfmt` over all code
+  rustfmt:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - run: rustup update stable && rustup default stable
+    - run: rustup component add rustfmt
+    - run: cargo fmt --all -- --check
+    - run: cd crates/cargo-test-macro && cargo fmt --all -- --check
+    - run: cd crates/cargo-test-support && cargo fmt --all -- --check
+    - run: cd crates/crates-io && cargo fmt --all -- --check
+    - run: cd crates/resolver-tests && cargo fmt --all -- --check
+    - run: cd crates/cargo-platform && cargo fmt --all -- --check
+
+  test:
+    runs-on: ${{ matrix.os }}
+    env:
+      CARGO_PROFILE_DEV_DEBUG: 1
+      CARGO_PROFILE_TEST_DEBUG: 1
+      CARGO_INCREMENTAL: 0
+    strategy:
+      matrix:
+        include:
+        - os: ubuntu-latest
+          rust: stable
+          other: i686-unknown-linux-gnu
+        - os: ubuntu-latest
+          rust: beta
+          other: i686-unknown-linux-gnu
+        - os: ubuntu-latest
+          rust: nightly
+          other: i686-unknown-linux-gnu
+        - os: macos-latest
+          rust: stable
+          other: x86_64-apple-ios
+        - os: windows-latest
+          rust: stable-msvc
+          other: i686-pc-windows-msvc
+        - os: windows-latest
+          rust: beta-gnu
```

Comment:
Curious, why switch to beta from nightly for Windows-gnu?

Reply:
At a technical level it was because I was lazy. One of the checks below checked for `rust == 'nightly'` for whether or not to install extra stuff, but if this used nightly-gnu I'd have to change that to "if `rust` starts with 'nightly'" as a condition, which I didn't trivially know how to do.

Otherwise though I believe this started on nightly because that was where it worked (changes with msys2 and which compilers are used). Now that the support is riding the trains I figured we'd want to let this move back to stable like everything else.

Reply:
One of my intents was to ensure that all the "is_nightly" tests get some coverage on Windows (indefinitely).  I think it substantially increases the test coverage for a platform that can be troublesome.

I can test it if you want, but it looks like there is a [startsWith](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#startswith).